### PR TITLE
Enable CPU Feature Masks

### PR DESCRIPTION
**Pull request by harryyu1994**
Signed-off-by: Harry Yu <harryyu1994@gmail.com>

### CODE REVIEW

**Comment by vijaysun-omr** [comment]
@genie-omr build all

File: compiler/env/OMRCPU.hpp [hunk 1]
```diff
@@ -107,13 +117,19 @@ class OMR_EXTENSIBLE CPU
...
     * @brief Returns the processor type and features that will be used by portable AOT compilations
     * @param[in] omrPortLib : the port library
     * @return TR::CPU
     */
    static TR::CPU detectRelocatable(OMRPortLibrary * const omrPortLib);
 
+   /**
+    * @brief Intialize _featureMasks to the list of processor features that will be utilized by the compiler and set _isFeatureMasksEnabled to true
+    * @return void
+    */
```
Comment (fjeremic) [inline compiler/env/OMRCPU.hpp:130]: Setting of `_isFeatureMasksEnabled` seems like an implementation detail. I don't think we should document this on an external API.

File: compiler/env/OMRCPU.cpp [hunk 2]
```diff
@@ -147,6 +147,14 @@ OMR::CPU::self()
    return static_cast<TR::CPU*>(this);
    }
 
+void
+OMR::CPU::enableFeatureMasks()
+   {
+   // Assume all features will be utilized by default
+   memset(_supportedFeatureMasks.features, ~0, OMRPORT_SYSINFO_FEATURES_SIZE*sizeof(uint32_t));
```
Comment (fjeremic) [inline compiler/env/OMRCPU.cpp:154]: This seems dangerous. Would it not be safer to assume they're all off instead?

File: compiler/env/OMRCPU.hpp [hunk 3]
```diff
@@ -107,13 +117,19 @@ class OMR_EXTENSIBLE CPU
...
+   static void enableFeatureMasks();
```
Comment (fjeremic) [inline compiler/env/OMRCPU.hpp:131]: I'm not quite understanding the purpose of this API. The commit does not explain it either. I think we need a better commit description.

**Pull request closed without merging**